src/main/java/lib/motors/TalonFXSim.java

```diff
@@ -0,0 +1,110 @@
+package lib.motors;
+
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.controls.*;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import lib.math.differential.Derivative;
+import lib.units.Units;
+
+public class TalonFXSim {
+
+    private final DCMotorSim motorSim;
+    private final Derivative acceleration = new Derivative();
+    private PIDController controller = null;
+    private ProfiledPIDController profiledController = null;
+    private double lastTimestampSeconds = 0;
+
+    private double voltageRequest;
+
+    public TalonFXSim(int numMotors, double gearing, double jKgMetersSquared) {
+        DCMotor motor = DCMotor.getFalcon500(numMotors);
```

Comment:
This should be supplied as a ctor parameter because other Talon FX motors exist (Kraken X60).

Reply:
Which we will not buy, but I get what you mean

Reply:
Maybe currently no, but in the future we probably will. Remember, the command library should be usable going years forward.

Reply:
2a05c56b7c57ec59cb3b1196a890449811db656d